packages/find-root/src/index.ts

```diff
@@ -71,6 +93,10 @@ export async function findRoot(cwd: string): Promise<MonorepoRoot> {
     return monorepoRoot;
   }
 
+  if (!toolsToInclude.includes(RootTool)) {
+    throw new NoMatchingMonorepoFound(cwd);
+  }
```

Comment:
I'm not entirely sure without actually testing this further - but I think that this actually won't work. Imagine this

```js
/*
  packages/package-a/package.json
  package.json
  pnpm-lock.yaml
  pnpm-workspace.yaml
*/
findRoot(pkgA, {
  tools: [PnpmTool, RootTool]
})
```

If I'm not mistaken - this would determine `packages/package-a` based on the `RootTool` being included in the list but the intention of the caller was clearly to find the pnpm root here.

Calling this properly, while allowing `RootTool`, would require something like this:
```ts
const root = (await findRoot(pkgA, { tools: [PnpmTool] })) || (await findRoot(pkgA, { tools: [RootTool] }))
```

Reply:
Ah, so it actually doesn't work like that since `RootTool.isMonorepoRoot` always returns `false` and we just have a special path for it in the `findRoot`. 

Disregard that comment, I also missed that this PR adds `RootTool` to the `DEFAULT_TOOLS`